src/bin/help.ts

```diff
@@ -0,0 +1,182 @@
+			}: ${description}`,
+				),
+			);
+		}
+	}
+}
+
+function createHelpTextSections(options: object): HelpTextSection[] {
+	const helpTextSections: HelpTextSection[] = [];
+
+	const core: HelpTextSection = {
+		sectionHeading: "Core options:",
+		subsections: [
+			{
+				flags: [],
+				subheading: undefined,
+				warning: undefined,
+			},
+		],
+	};
+
+	const optional: HelpTextSection = {
+		sectionHeading: "Optional options:",
+		subsections: [
+			{
+				flags: [],
+				subheading: undefined,
+				warning: undefined,
+			},
+		],
+	};
+
+	const optOut: HelpTextSection = {
+		sectionHeading: "Opt-outs:",
+		subsections: [
+			{
+				flags: [],
+				subheading: undefined,
+				warning: `
+      ⚠️ Warning: Specifying any --exclude-* flag on the command-line will 
+      cause the setup script to skip prompting for more excludes. ⚠️`,
+			},
+			{
+				flags: [
+					{
+						description: `Skips network calls that fetch all-contributors
+        data from GitHub`,
+						flag: "exclude-contributors",
+						type: "boolean",
+					},
+				],
+				subheading: `
+      You can prevent the migration script from making some network-based 
+      changes using any or all of the following CLI flags:`,
+				warning: undefined,
+			},
+			{
+				flags: [],
+				subheading: `
+      You can prevent the migration script from making some changes on disk 
+      using any or all of the following CLI flags:`,
+				warning: undefined,
+			},
+		],
+	};
+
+	for (const option of Object.keys(options)) {
+		const data = options[option as keyof object] as unknown as Option;
+
+		if (data.docsSection === "core") {
+			core.subsections[0].flags.push({
+				description: data.description,
+				flag: option,
+				type: data.type,
+			});
+		}
+
+		if (data.docsSection === "optional") {
+			optional.subsections[0].flags.push({
+				description: data.description,
+				flag: option,
+				type: data.type,
+			});
+		}
+
+		if (data.docsSection === "opt-out") {
+			optOut.subsections[0].flags.push({
+				description: data.description,
+				flag: option,
+				type: data.type,
+			});
+		}
+
+		if (data.docsSection === "skip-net") {
+			optOut.subsections[1].flags.push({
+				description: data.description,
+				flag: option,
+				type: data.type,
+			});
+		}
+
+		if (data.docsSection === "skip-disk") {
+			optOut.subsections[2].flags.push({
+				description: data.description,
+				flag: option,
+				type: data.type,
+			});
+		}
+	}
+
+	helpTextSections.push(core, optional, optOut);
+
+	return helpTextSections;
+}
+
+export function logHelpText(): void {
+	const helpTextSections = createHelpTextSections(allArgOptions);
+
+	logPipelessLine();
+
+	logPipelessLine(
+		chalk.cyan(
+			`
+      A quickstart-friendly TypeScript template with comprehensive formatting, 
+      linting, releases, testing, and other great tooling built-in.
+      `,
+		),
+	);
+
+	for (const section of helpTextSections) {
+		logHelpTextSection(section);
+
+		logPipelessLine();
+	}
+}
```

Comment:
```plaintext
┌  ✨ Welcome to create-typescript-app  1.29.39! ✨
│
│  ⚠️ This template is early stage, opinionated, and not endorsed by the TypeScript team. ⚠️
│  ⚠️ If any tooling it sets displeases you, you can always remove that portion manually. ⚠️
 

      A quickstart-friendly TypeScript template with comprehensive formatting, 
      linting, releases, testing, and other great tooling built-in.
      
 
   Core options:

      --base (string): Whether to scaffold the repository with:
        • everything: that comes with the template (recommended)
        • minimum: amounts of tooling, essentially opting out of everything
        • prompt: for which portions to exclude

      --create-repository: Whether to create a corresponding repository on 
        github.com (if it doesn't yet exist)
```

![Screenshot of the previous code block in a light mode terminal](https://github.com/JoshuaKGoldberg/create-typescript-app/assets/3335181/f2015852-503b-4981-98e6-05832eb1073f)

![Screenshot of the previous code block in a dark mode terminal](https://github.com/JoshuaKGoldberg/create-typescript-app/assets/3335181/d085c904-715b-47d6-aa4e-bf24481ae463)

[Cleanup] A few thoughts on the display here:

* The `│` / `┌` piping is inconsistent and IMO looks a little off as a result. I think that within a single session, it should either always use it or never.
* The rest of lines are super indented but don't need to be. If there's no `│` piping then let's not add those extra indents?
* What do you think about bolding the options (here, `--base`) to make them stand out?
* I don't see `chalk.cyanBright` any differently (Mac, VS Code, zsh). Maybe `blueBright` instead to make them stand out more?
* After the lists of options, maybe it could log something like `See more documentation at https://github.com/JoshuaKGoldberg/create-typescript-app. 💖`? Some way to encourage folks to visit the repo?

Reply:
@dominicduffin1 bump on this thread?